public/app/features/dashboard/components/PanelEditor/getVisualizationOptions.tsx

```diff
@@ -240,8 +244,8 @@ export function getVisualizationOptions2(props: OptionPaneRenderProps2): Options
     const hideOption =
       fieldOption.showIf &&
       (fieldOption.isCustom
-        ? !fieldOption.showIf(currentFieldConfig.defaults.custom, data?.series)
-        : !fieldOption.showIf(currentFieldConfig.defaults, data?.series));
+        ? !fieldOption.showIf(currentFieldConfig.defaults.custom, data?.series, data?.annotations)
```

Comment:
I'm not sure if we have any tests for this. If not maybe you could add ...? 🥺

Reply:
Done! The test file diff is a lot but only because it was only testing one function, so I had to change the structure to wrap 2.